03-events-forms-refs-life-cycle/forms-and-refs/todo-list/src/TodoList.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import Todo from './Todo';
+import NewTodoForm from './NewTodoForm';
+import EditTodoForm from "./EditTodoForm";
+import "./TodoList.css"
+
+class TodoList extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      todos: []
+    }
+  }
+
+  completeItem(key) {
+    if (this.state.todos[key].status === "notDone") {
+      let newTodos = this.state.todos.slice();
+      newTodos[key].status = "done";
+      this.setState( {todos:newTodos} );
+    } else {
+      let newTodos = this.state.todos.slice();
+      newTodos[key].status = "notDone";
+      this.setState( {todos:newTodos} );
+    }
+  }
+
+  deleteItem(key) {
+    if (this.state.todos[key].status === "done"){
+      let newTodos = this.state.todos.slice();
+      newTodos.splice(key, 1);
+      this.setState( {todos:newTodos} );
+    }
+  }
+
+  starItem(key) {
+    if (this.state.todos[key].type === null) {
```

Comment:
If I do that, how can I tell my TodoList which button has triggered the new combined function? It would either be completing, starring, or deleting a Todo item. I've started updating my code if you want to see my thinking. My problem is that since I've bound the keyword `this` so that it's the TodoList component, I'm not sure how to access the onClick event itself.